pjmedia/src/pjmedia/sdp_neg.c

```diff
@@ -1793,3 +1946,167 @@ PJ_DEF(pj_status_t) pjmedia_sdp_neg_fmt_match(pj_pool_t *pool,
                             offer, o_fmt_idx, answer, a_fmt_idx, option);
 }
 
+static unsigned assign_new_pt(pj_pool_t *pool, media_pt_map *used_pt, 
+                              const pj_str_t *codec)
+{
+    unsigned pt_check;
+    unsigned i = 0;
+    unsigned used_pt_num = used_pt->map_cnt;
+    unsigned max_pt_cnt = PJ_ARRAY_SIZE(used_pt->map);
+    const pj_str_t telephone_event = pj_str("telephone-event");
+
+    if (pj_strncmp(codec, &telephone_event, telephone_event.slen) == 0)
+        pt_check = PJMEDIA_RTP_PT_TELEPHONE_EVENTS;
+    else
+        pt_check = START_DYNAMIC_PT;
+    
+    for (; i < used_pt_num && i < max_pt_cnt; ++i) {
+        if (pj_strcmp(codec, &used_pt->map[i].codec) == 0) {
+            return used_pt->map[i].pt;
```

Comment:
Ops, still the new name does not seem to represent the function action, i.e: it looks like the function lookups PT for a given codec, and assign if not found. Well, perhaps better put description what the function really does :)